### PR TITLE
deprecating overloads makeCRAMWriter with mixed Path/File

### PR DESCRIPTION
**Pull request by lbergelson**
* deprecating overloads of SAMFileWriterFactory.makeCRAMWriter that took a mix of Path and File as inputs
   these were needed when we supported Path for the CRAM but not the reference

* removing one overload that seems redundant and hasn't been in any htsjdk release yet

### CODE REVIEW

**Comment by lbergelson** [comment]
@cmnbroad Since you brought up the issue I'm going to let you review this one.
@magicDGS any objections to this change?

**Comment by lbergelson** [comment]
@cmnbroad What was the comment?

File: src/main/java/htsjdk/samtools/SAMFileWriterFactory.java [hunk 1]
```diff
@@ -550,9 +550,11 @@ public CRAMFileWriter makeCRAMWriter(final SAMFileHeader header, final File outp
      * @param referenceFasta reference sequence file
      * @return CRAMFileWriter
      *
+     * @deprecated since 6/18, use {@link #makeCRAMWriter(SAMFileHeader, boolean, Path, Path)} instead
```
Comment (cmnbroad) [inline src/main/java/htsjdk/samtools/SAMFileWriterFactory.java]: @lbergelson Hm I dropped my connection. Anyway, should we suggest migration to `makeWriter(SAMFileHeader, boolean, Path, Path)` as the canonical all-knowing method. And someday deprecate all the rest ?

Comment (lbergelson) [reply src/main/java/htsjdk/samtools/SAMFileWriterFactory.java]: Oh, you're right.  I'll change.

**Comment by lbergelson** [comment]
@cmnbroad Is this better?

**Comment by magicDGS** [comment]
No objections to this @lbergelson - I did not want to remove or not include the overrides to do not break backwards compatibility. Deprecation sounds like a good solution!

**Pull request merged**